src/main/java/org/terasology/gooeyDefence/economy/ShopManager.java

```diff
@@ -15,108 +15,46 @@
 import org.terasology.gooeyDefence.DefenceUris;
 import org.terasology.gooeyDefence.events.OnFieldReset;
 
-import java.util.HashSet;
 import java.util.Optional;
-import java.util.Set;
-import java.util.stream.Collectors;
 
 /**
  * Handles the purchasing of blocks
  */
 @RegisterSystem
-@Share(ShopManager.class)
 public class ShopManager extends BaseComponentSystem {
-
-    private Set<Block> purchasableBlocks = new HashSet<>();
-    private Set<Prefab> purchasableItems = new HashSet<>();
-
     @In
     private AssetManager assetManager;
     @In
-    private BlockManager blockManager;
-    @In
-    private InventoryManager inventoryManager;
-    @In
     private LocalPlayer localPlayer;
     @In
     private EntityManager entityManager;
 
-    private BlockItemFactory blockItemFactory;
-
-    /**
-     * Gets how much money a ware will cost.
-     * Tries to use the cost on the purchasable component, with the value component as a fallback.
-     *
-     * @param ware The ware to get the price for
-     * @return The price of the ware.
-     */
-    public static int getWareCost(ComponentContainer ware) {
-        int cost = ware.getComponent(PurchasableComponent.class).cost;
-        if (cost < 0) {
-            if (ware.hasComponent(ValueComponent.class)) {
-                return ware.getComponent(ValueComponent.class).value;
-            } else {
-                return 0;
-            }
-        } else {
-            return cost;
-        }
-    }
-
-    @Override
-    public void postBegin() {
-        blockItemFactory = new BlockItemFactory(entityManager);
-        BlockExplorer blockExplorer = new BlockExplorer(assetManager);
+    @ReceiveEvent(priority = EventPriority.PRIORITY_LOW)
+    public void onPlayerJoin(OnPlayerSpawnedEvent onPlayerSpawnedEvent, EntityRef player) {
+        CurrencyStorageComponent component = new CurrencyStorageComponent();
+        component.amount = 100;
 
-        purchasableItems = assetManager.getLoadedAssets(Prefab.class)
-                .stream()
-                .filter(prefab -> prefab.hasComponent(ItemComponent.class)
-                        && prefab.hasComponent(PurchasableComponent.class))
-                .collect(Collectors.toSet());
-
-        Set<BlockUri> blocks = new HashSet<>();
-        blocks.addAll(blockManager.listRegisteredBlockUris());
-        blocks.addAll(blockExplorer.getAvailableBlockFamilies());
-        blocks.addAll(blockExplorer.getFreeformBlockFamilies());
-
-        purchasableBlocks = blocks.stream()
-                .map(blockManager::getBlockFamily)
-                .map(BlockFamily::getArchetypeBlock)
-                .filter(block -> block.getPrefab().isPresent())
-                .filter(block -> block.getPrefab().get().hasComponent(PurchasableComponent.class))
-                .collect(Collectors.toSet());
+        player.addOrSaveComponent(component);
+        player.send(new WalletUpdatedEvent(component.amount));
```

Comment:
Ah, I see - why was this change necessary? (You probably already explained that before... :see_no_evil: )